### PR TITLE
Update how Anglerfish is applied

### PR DESCRIPTION
**Pull request by kedhammar**

### CODE REVIEW

**Comment by kedhammar** [comment]
related https://github.com/SciLifeLab/scilifelab_epps/pull/304

File: taca/nanopore/ONT_run_classes.py [hunk 1]
```diff
@@ -459,12 +459,33 @@ def fetch_anglerfish_samplesheet(self) -> bool:
...
+            
+    def has_fastq_output(self) -> bool:
+        """Check whether run has fastq output."""
+
+        reads_dir = os.path.join(self.run_abspath, "fastq_pass")
+
+        if os.path.exists(reads_dir):
+            return True
+        else:
+            return False
```
Comment (alneberg) [inline taca/nanopore/ONT_run_classes.py:471]: Some old-school hacker hack
```suggestion
        return os.path.exists(reads_dir)
```

File: taca/nanopore/ONT_run_classes.py [hunk 2]
```diff
@@ -475,31 +496,46 @@ def run_anglerfish(self):
...
             "conda run -n anglerfish " + " ".join(anglerfish_command),
             # Dump Anglerfish exit code into file
             f"echo $? > {self.anglerfish_done_abspath}",
-            # Copy the Anglerfish samplesheet used to start the run into the run dir, for traceability
-            # (The correct anglerfish run dir is identified by it being younger than the "run-ongoing" file)
-            f"new_runs=$(find . -type d -name 'anglerfish_run*' -newer {self.anglerfish_ongoing_abspath})",
-            f"if [[ $(echo '${{new_runs}}' | wc -l) -eq 1 ]] ; then cp {self.anglerfish_samplesheet} ${{new_runs}}/ ; fi",
-            # Regardless of exit status: Remove 'run-ongoing' file.
+            # Move run to subdir
+            f'find {self.run_abspath} -name "anglerfish_run*" -type d -newer {self.run_abspath}/.anglerfish_ongoing '
```
Comment (alneberg) [inline taca/nanopore/ONT_run_classes.py:520]: I was struggling to understand this command. Could you keep the old comments and perhaps correct them if you think they are not correct anymore?

File: taca/nanopore/ONT_run_classes.py [hunk 3]
```diff
@@ -475,31 +496,46 @@ def run_anglerfish(self):
...
+
         # Start Anglerfish subprocess
-        process = subprocess.Popen(
-            "; ".join(full_command),
-            shell=True,
-            cwd=self.run_abspath,
-        )
+        with open(stderr_relpath, 'w') as stderr:
+            process = subprocess.Popen(
+                "; ".join(full_command),
```
Comment (alneberg) [inline taca/nanopore/ONT_run_classes.py:534]: Could we even run the command.sh file?

Comment (kedhammar) [reply taca/nanopore/ONT_run_classes.py]: [0baa97c](https://github.com/SciLifeLab/TACA/pull/409/commits/0baa97c9898eb71061b8b2ac2b86b44e52de2c20)

File: taca/nanopore/ONT_run_classes.py (hunk 2 above)
Comment (kedhammar) [reply taca/nanopore/ONT_run_classes.py:519]: [0baa97c](https://github.com/SciLifeLab/TACA/pull/409/commits/0baa97c9898eb71061b8b2ac2b86b44e52de2c20)

**Comment by kedhammar** [comment]
Tested on preproc, performs as expected

**Review by alneberg — APPROVED** [review]
Looks good!

**Pull request merged**